src/main/webapp/view/transcript/transcript.js

```diff
@@ -171,3 +173,43 @@ function scrollToTopOfTranscript(transcriptLine) {
   const ulElementOffset = transcriptLine.parentElement.offsetTop;
   transcriptContainer.scrollTop = transcriptLine.offsetTop - ulElementOffset;
 }
+
+/**
+ * Returns the next transcript line based on `currentTimeMs`.
+ */
+function getNextTranscriptLine(currentTimeMs) {
+  const nextTranscript = currentTranscriptLine.nextElementSibling;
+  // Video is playing normally.
+  if (isWithinTimeRange(currentTimeMs, nextTranscript)) {
+    return nextTranscript;
+  }
+  return findClosestTranscriptLine(currentTimeMs);
+}
+
+/**
+ * Searches for and returns the closest transcript line
+ * based on `timeMs`.
+ */
+function findClosestTranscriptLine(timeMs) {
+  // TODO: Create a global variable for the list of transcript line elements
+  // once the pull request separating transcript.js into classes is merged.
+  const transcriptLineElements = document.getElementsByTagName('li');
+  let transcriptLinePointer = transcriptLineElements[0];
+  while (transcriptLinePointer != null &&
+         !isWithinTimeRange(timeMs, transcriptLinePointer) &&
+         isBeforeTimeMs(timeMs, transcriptLinePointer)) {
+    transcriptLinePointer = transcriptLinePointer.nextElementSibling;
+  }
+  if (transcriptLinePointer === null) {
+    transcriptLinePointer =
+        transcriptLineElements[transcriptLineElements.length - 1];
+  }
```

Comment:
can you add a comment about this transcript line if-statement and what it means. Seems like it is doing something special.

Reply:
done, thanks!